pkg/shp/flags/build.go

```diff
@@ -60,48 +56,62 @@ func BuildSpecFromFlags(flags *pflag.FlagSet) *buildv1alpha1.BuildSpec {
 	imageAnnotationsFlags(flags, spec.Output.Annotations)
 	buildRetentionFlags(flags, spec.Retention)
 
-	return spec
+	var dockerfile, builderImage string
+	dockerfileFlags(flags, &dockerfile)
+	builderImageFlag(flags, &builderImage)
+
+	return spec, &dockerfile, &builderImage
 }
 
 // SanitizeBuildSpec checks for empty inner data structures and replaces them with nil.
-func SanitizeBuildSpec(b *buildv1alpha1.BuildSpec) {
+func SanitizeBuildSpec(b *buildv1beta1.BuildSpec) {
 	if b == nil {
 		return
 	}
-	if b.Source.Credentials != nil && b.Source.Credentials.Name == "" {
-		b.Source.Credentials = nil
-	}
-	if b.Source.ContextDir != nil && *b.Source.ContextDir == "" {
-		b.Source.ContextDir = nil
-	}
-	if b.Source.Revision != nil && *b.Source.Revision == "" {
-		b.Source.Revision = nil
-	}
-	if b.Source.URL != nil && *b.Source.URL == "" {
-		b.Source.URL = nil
-	}
-	if b.Source.BundleContainer != nil && b.Source.BundleContainer.Image == "" {
-		b.Source.BundleContainer = nil
-	}
-	if b.Builder != nil {
-		if b.Builder.Credentials != nil && b.Builder.Credentials.Name == "" {
-			b.Builder.Credentials = nil
+
+	if b.Source != nil {
```

Comment:
I am missing the code which is setting source.type.

Reply:
Here, I am setting the type : https://github.com/shipwright-io/cli/blob/fa0e5372c37447feaa821e48c45132d391069009/pkg/shp/cmd/build/create.go#L69